python/pyarrow/tests/test_substrait.py

```diff
@@ -605,3 +605,151 @@ def table_provider(names, schema):
     expected = pa.Table.from_pydict({"out": [1, 2, 3]})
 
     assert res_tb == expected
+
+
+def test_aggregate_udf_basic(varargs_agg_func_fixture):
```

Comment:
This test case matches how we would this it internally. The end user would define sth like
```
def foo(v: pd.Series):
    return np.nanmean(v)

summarize(table, agg=foo, columns=['v'], by='time')
```

We would then wrap the foo into a function that Acero is expecting (a varargs UDF)
```
def get_acero_func(func):
      def acero_func(ctx, *args):
            return pa.scalar(func(*[arg.to_pandas() for arg in args]))
```
And also register it in Acero on the fly.